cuenca/http/client.py

```diff
@@ -137,6 +115,31 @@ def request(
         self._check_response(resp)
         return resp.json()
 
+    def set_jwt_headers(self):
+        # Make sure the current token is still valid
+        try:
+            payload_encoded = self.jwt_token.split('.')[1]
+            payload = json.loads(base64.b64decode(f'{payload_encoded}=='))
+            # Get a new token if there's less than 5 mins for the actual
+            # to be expired
+            if dt.datetime.utcfromtimestamp(
+                payload['exp']
+            ) - dt.datetime.utcnow() <= dt.timedelta(minutes=5):
+                raise Exception('Expired token')
+        except Exception:
```

Comment:
There are few cases in life where you should catch `Exception`. Why are you doing it in this case?

Reply:
There're a few cases here where this part of the code can generate an `Exception`:

- `IndexError` if the current token is malformed
- `binascii.Error` if the token can't be decoded with base64
- `JSONDecodeError` if the JSON is malformed
- I added one if the token is expired

All of this exceptions can happen if `authed` responded with an invalid token or the current token is somehow manipulated (which there should not be a reason you would want to do it). In any case, the token should be generated again.

If I include only one line of code per `try` block, I'd ended up having something like:
```python
        try:
            payload_encoded = self.jwt_token.split('.')[1]
        except IndexError:
             self.jwt_token = None
       else:
           try:
                 payload = json.loads(base64.b64decode(f'{payload_encoded}=='))
           except (binascii.Error, JSONDecodeError):
               self.jwt_token = None
           else:
                if dt.datetime.utcfromtimestamp(
                    payload['exp']
                ) - dt.datetime.utcnow() <= dt.timedelta(minutes=5):
                    self.jwt_token = None
```

Which I think is way more confusing. I didn't see it necessary to create a new `Exception` since it's only going to be used here but I could clean it up by specify all the possible exceptions:
```python
        try:
            payload_encoded = self.jwt_token.split('.')[1]
            payload = json.loads(base64.b64decode(f'{payload_encoded}=='))
            # Get a new token if there's less than 5 mins for the actual
            # to be expired
            if dt.datetime.utcfromtimestamp(
                payload['exp']
            ) - dt.datetime.utcnow() <= dt.timedelta(minutes=5):
                raise CustomException('Expired token')
        except (IndexError, binascii.Error, JSONDecodeError, CustomException):
            self.jwt_token = None
```